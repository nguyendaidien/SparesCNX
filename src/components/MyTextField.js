import { Field } from 'formik'
import { TextField } from '@material-ui/core'
import PropTypes from 'prop-types'
import React from 'react'

function MyTextField(textFieldProps) {
  return <Field {...textFieldProps} component={FormikTextField} />
}

function FormikTextField({ field, form, onBlur, onChange, ...textFieldProps }) {
  function onChangeWrapper(e) {
    field.onChange(e)
    if (onChange) {
      onChange(e)
    }
  }

  function onBlurWrapper(e) {
    field.onBlur(e)
    if (onBlur) {
      onBlur(e)
    }
  }

  return (
    <TextField
      style={{ width: '100%' }}
      label={field.name}
      variant='outlined'
      id={field.name}
      name={field.name}
      value={field.value}
      onChange={onChangeWrapper}
      onBlur={onBlurWrapper}
      {...textFieldProps}
    />
  )
}

export default MyTextField

MyTextField.propTypes = {
  type: PropTypes.string,
  name: PropTypes.string,
  label: PropTypes.string,
  disabled: PropTypes.bool
}
