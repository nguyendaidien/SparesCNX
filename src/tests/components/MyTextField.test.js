import React from 'react';
import { shallow } from 'enzyme';
import MyTextField from '../../components/MyTextField';

test('should test MyTextField component', () => {
    const wrapper = shallow(<MyTextField />);
    expect(wrapper).toMatchSnapshot();
});