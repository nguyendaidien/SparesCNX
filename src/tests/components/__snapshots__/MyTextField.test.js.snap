// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should test MyTextField component 1`] = `ShallowWrapper {}`;
